general/mem_manager.hpp

```diff
@@ -76,7 +112,7 @@ MemoryClass operator*(MemoryClass mc1, MemoryClass mc2);
 
     A Memory object stores up to two different pointers: one host pointer (with
     MemoryType from MemoryClass::HOST) and one device pointer (currently one of
-    MemoryType::CUDA or MemoryTyep::CUDA_UVM).
+    MemoryType: DEVICE, DEVICE_UMPIRE, DEVICE_UVM or DEVICE_DEBUG).
```

Comment:
Currently there is no type `DEVICE_UVM`.

Reply:
`Updated`